frontend/src/components/object/ObjectFileDetails.vue

```diff
@@ -29,32 +46,88 @@ const getObjectInfo = async (objId: string) => {
   }
 };
 
+const showPermissions = async (objectId: string, objectName: string) => {
+  permissionsVisible.value = true;
+  permissionsObjectId.value = objectId;
+  permissionsObjectName.value = objectName;
+};
+
 onMounted(() => {
   getObjectInfo(route.query.objId as string);
 });
 </script>
 
 <template>
-  <div class="flex justify-content-start">
-    <div class="flex col align-items-center pl-0">
-      <font-awesome-icon
-        icon="fa-solid fa-circle-info"
-        style="font-size: 2rem"
-      />
-      <h1 class="pl-1 font-bold">
-        File details
-      </h1>
+  <div>
+    <div class="flex justify-content-start">
+      <div class="flex col align-items-center pl-0">
+        <font-awesome-icon
+          icon="fa-solid fa-circle-info"
+          style="font-size: 2rem"
+        />
+        <h1 class="pl-1 font-bold">
+          File details
+        </h1>
+      </div>
+      <div class="action-buttons">
+        <Button class="p-button-lg p-button-text">
+          <font-awesome-icon icon="fa-solid fa-share-nodes" />
+        </Button>
+        <DownloadObjectButton
+          v-if="objectStore.isActionAllowed(objectInfo.permissions, Permissions.READ, currentUser?.userId)"
+          :mode="ButtonMode.ICON"
+          :ids="[objectInfo.id]"
+        />
+        <Button
+          v-if="objectStore.isActionAllowed(objectInfo.permissions, Permissions.MANAGE, currentUser?.userId)"
+          class="p-button-lg p-button-text"
+          @click="showPermissions(objectInfo.id, objectInfo.name)"
+        >
+          <font-awesome-icon icon="fa-solid fa-users" />
+        </Button>
+        <DeleteObjectButton
+          v-if="objectStore.isActionAllowed(objectInfo.permissions, Permissions.DELETE, currentUser?.userId)"
+          :mode="ButtonMode.ICON"
+          :ids="[objectInfo.id]"
+        />
+      </div>
+    </div>
+
+    <div
+      v-if="isInfoLoaded"
+      class="pl-2"
+    >
+      <ObjectProperties :object-properties="objectInfo" />
```

Comment:
you could rename these props to `object-info="objectInfo"` so that they are received in the child component as 
`objectInfo`